File: backend/src/main/java/com/me/ecommerce/entity/Product.java
package com.me.ecommerce.entity;

//import jakarta.persistence;


import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.io.Serial;
import java.util.LinkedHashSet;
import java.util.Set;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import java.math.BigDecimal;
import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "product")
public class Product extends Audit {
    @Serial
    private static final long serialVersionUID = -6188767059145560467L;

    @Id
    @GeneratedValue(generator = "UUID")
    @Column(name = "id", nullable = false)
    private UUID id;

    @Column(name = "sku")
    private String sku;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "unit_price", precision = 19, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "image_url")
    private String imageUrl;

    @ColumnDefault("true")
    @Column(name = "active")
    private boolean active;

    @Column(name = "units_in_stock")
    private int unitsInStock;

    @Column(name = "units_sold")
    private Integer unitsSold;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    private ProductCategory category;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<OrderItem> orderItems = new LinkedHashSet<>();

    public void addOrderItem(OrderItem orderItem) {
        if (orderItem != null) {
            if (orderItems == null) {
                orderItems = new LinkedHashSet<>();
            }
            orderItems.add(orderItem);
            orderItem.setProduct(this);
        }
    }

}

File: frontend/src/app/interfaces/product.ts
export interface Product {
  id?: string;
  sku?: string;
  name?: string;
  description?: string;
  unitPrice?: number;
  imageUrl?: string;
  active?: boolean;
  unitsInStock?: number;
  unitsSold?: number;
  createdAt?: Date;
  updatedAt?: Date;
}





// export class Product {
//
//   constructor(
//     public id?: string,
//     public sku?: string,
//     public name?: string,
//     public description?: string,
//     public unitPrice?: number,
//     public imageUrl?: string,
//     public active?: boolean,
//     public unitsInStock?: number,
//     public unitsSold?: number,
//     public createdAt?: Date,
//     public updatedAt?: Date
//   ) {
//     // You can also initialize fields here if needed
//   }
// }
